### PR TITLE
Build ILCompiler NuGet packages

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
This is using the same strategy we used to build the package in the runtimelab repo. It's meant to start conversation.

I'll tag the infra team after the new year. Just sending out to see how the CI takes this.

### CODE REVIEW

**Comment by tomrus88** [comment]
Any news on this?

**Comment by MichalStrehovsky** [comment]
> Any news on this?

We still publish pacakges from runtimelab so I kind of forgot about this.

@safern @jkoritzinsky would you have the most context on packaging to be able to review this?

File: src/installer/pkg/projects/nativeaot-packages.proj [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<Project>
+  <Import Sdk="Microsoft.Build.Traversal" Project="Sdk.props" />
+
+  <ItemGroup>
+    <Project Include="Microsoft.DotNet.ILCompiler\Microsoft.DotNet.ILCompiler.pkgproj" />
+    <ProjectReference Include="@(Project)" />
+    <ProjectReference Include="@(Project)" AdditionalProperties="PackageTargetRuntime=$(PackageRID)" />
```
Comment (safern) [inline src/installer/pkg/projects/nativeaot-packages.proj:7]: Why do we need to include it once without the additional properties and then with it?

Comment (MichalStrehovsky) [reply src/installer/pkg/projects/nativeaot-packages.proj:7]: One builds the "lineup package" (or what's the right term - the one without the `runtime.$RID` prefix), the other builds the RID-specific package (with the `runtime.$RID` prefix).

Comment (safern) [reply src/installer/pkg/projects/nativeaot-packages.proj:7]: Ah cool.

**Comment by MichalStrehovsky** [comment]
@safern do you have any objections to merging this?

File: src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  <PropertyGroup>
+    <GeneratePackage>true</GeneratePackage>
+    <IncludeSymbolsInPackage>true</IncludeSymbolsInPackage>
+    <PackageDescription>Provides a native AOT compiler and runtime for .NET</PackageDescription>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <!-- For now, send symbols package into a dummy directory to avoid conflicts with dotnet/runtime and
+         to avoid issues with objwriter not having the rigth ELF version resource -->
+    <SymbolPackageOutputPath>$([MSBuild]::NormalizeDirectory('$(ArtifactsPackagesDir)', 'Ignored'))</SymbolPackageOutputPath>
```
Comment (safern) [inline src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj:12]: If we send them into `NonShipping` do them also get uploaded to symbol server? @mmitche do you know?

File: src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  </PropertyGroup>
+
+  <Target Name="GetIlcCompilerFiles"
+          DependsOnTargets="ResolveLibrariesFromLocalBuild"
+          BeforeTargets="GetPackageFiles">
+
+    <ItemGroup Condition="'$(PackageTargetRuntime)' != ''">
+      <File Include="@(LibrariesRuntimeFiles)">
+        <TargetPath>framework</TargetPath>
```
Comment (safern) [inline src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj:21]: Small stiling nit, this kind of metadata makes the file smaller and simpler to read if declared inline: i.e:

```xml
<File Include="@(LibrariesRuntimeFiles)" TargetPath="framework" />
```

**Review by safern — APPROVED** [review]
Other than 2 comments/questions LGTM

File: src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj (hunk 2 above)
Comment (MichalStrehovsky) [reply src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj]: I'll just try this without the SymbolPackageOutputPath - the concern about dotnet/runtime conflict in the comment no longer applies and the objwriter package is now built with Arcade's help.

Comment (mmitche) [reply src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj]: I **think** they do. But not positive.

Comment (mmitche) [reply src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj]: Might just do a quick official test (assign the build to General Testing) and see what happens.

**Comment by MichalStrehovsky** [comment]
Marking no merge because I'm seeing some fun failures in official builds.

**Comment by MichalStrehovsky** [comment]
Got a good official build and successful build promotion.

The libraries test failure is the known Windows bug (hang on process shutdown).

**Pull request merged**